### PR TITLE
New annotations “kubectl.kubernetes.io/default-container”

### PR DESCRIPTION
**Pull request by pacoxu**
https://github.com/kubernetes/enhancements/issues/2227#issuecomment-797269213

It is implemented in  1.21

### CODE REVIEW

File: content/en/docs/reference/labels-annotations-taints.md [hunk 1]
```diff
@@ -254,3 +254,8 @@ Example: `node.cloudprovider.kubernetes.io/shutdown:NoSchedule`
 
 If a Node is in a cloud provider specified shutdown state, the Node gets tainted accordingly with `node.cloudprovider.kubernetes.io/shutdown` and the taint effect of `NoSchedule`.
 
+## kubectl.kubernetes.io/default-container
+
+Example: `kubectl.kubernetes.io/default-container:container1`
+
+If a Pod has the annotation `kubectl.kubernetes.io/default-container`, consumers (CLI, operators) know which is the default container. For example, `kubectl logs` or `kubectl exec` without `-c` or `--container` flag will use this default container.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints.md:261]: 1. We're listing this in the wrong place (under a heading “**The taints listed below are always used on Nodes**”)
1. We should explicitly say that the value of the annotation is the container _name_ that is default for this Pod.

File: content/en/docs/reference/labels-annotations-taints.md [hunk 2]
```diff
@@ -254,3 +254,8 @@ Example: `node.cloudprovider.kubernetes.io/shutdown:NoSchedule`
 
 If a Node is in a cloud provider specified shutdown state, the Node gets tainted accordingly with `node.cloudprovider.kubernetes.io/shutdown` and the taint effect of `NoSchedule`.
 
+## kubectl.kubernetes.io/default-container
+
+Example: `kubectl.kubernetes.io/default-container:container1`
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints.md:259]: I'd pick a more plausible name and format it like other example annotations. For example:
```suggestion
Example: `kubectl.kubernetes.io/default-container: "front-end-app"`
```

**Comment by sftim** [comment]
LGTM
/approve

(would like someone else to /lgtm)

**Comment by rikatz** [comment]
/lgtm

From sig-cli perspective

Thank you @pacoxu !

**Pull request merged**